### PR TITLE
status: if spoiler_text is present, treat media as sensitive (#1010)

### PR DESCRIPTION
**Pull request by rtucker**
In the event that there is a CW, go ahead and treat any attached media as sensitive.

This addresses UI issue #1010

### CODE REVIEW

**Review by yiskah — APPROVED** [review]
This appears to be a needed non-breaking change. I don't see any syntax errors.

File: app/controllers/api/v1/statuses_controller.rb [hunk 1]
```diff
@@ -61,7 +61,7 @@ def create
                                          status_params[:status],
                                          status_params[:in_reply_to_id].blank? ? nil : Status.find(status_params[:in_reply_to_id]),
                                          media_ids: status_params[:media_ids],
-                                         sensitive: status_params[:sensitive],
+                                         sensitive: status_params[:spoiler_text].blank? ? status_params[:sensitive] : true,
```
Comment (Gargron) [inline app/controllers/api/v1/statuses_controller.rb]: Wouldn't it be easier to just have this line, without any others? Here you're already setting sensitive=true when spoiler_text=present. No reason to keep checking spoiler_text beyond this point.

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/statuses_controller.rb (hunk 1 above)
Comment (beatrix-bitrot) [inline app/controllers/api/v1/statuses_controller.rb]: mmm, the additional checks i guess cause this to apply retroactively as well as going forward though.

could be nice. if it doesn't do any harm let it go that way?

**Comment by nightpool** [comment]
I'm going to repost my objection from #1010:

> I believe I was the person @cassolotl mentioned, and yes, I believe that
keeping the affordance for hiding text content separate from the affordance
for hiding image content is a good thing. I think that there are multiple
use cases served by keeping them separate that unifying them would not.

> A good example is a longer discussion behind a cut, with images as an
attention grabber or hook. Another example is linking and citing sources
for an aesthetic post—they often take up multiple lines, so having them
behind a show more is preferable. Another one is the workaround already
mentioned.

I think that a much better way forward would have the JS *default* to having the media marked as sensitive if there is spoiler text present (both when you add the media to a post with existing spoiler text and when you add the spoiler text to existing media)

**Comment by rtucker** [comment]
I've been running this patch for a few days now, and upon further consideration, I think I agree with @nightpool -- I've seen a number of cases where the implied-NSFW was not appropriate.  I am going to close this pull request in favor of moving the implied-NSFW to the UI side of things.

**Pull request closed without merging**